ui: Release notes signup - Show Alert notification after signup

koorosh opened the pull request:
Resolves: #43912

Existing AlertBox component doesn't fit to new designs from Design system,
so new one is added as a wrapper on top of Ant design component.
The major problem which occur is how to integrate new AlertMessage component
to live with old one and do not affect each other behaviour.
To achieve this:
- AlertMessage component (new one) has the same set of properties. As result it will allow easily consume old alert messages by new component.
- To preserve old messages without any changes, one additional property is
added to specify which Alert component to show, old or new one.
- Current implementation reuses existing logic where we store state of alert in
local settings instead of redux store.

koorosh committed:
ui: Alert notification component

Existing AlertBox component doesn't fit to new designs from Design system,
so new one is added as a wrapper on top of Ant design component.
The major problem which occur is how to integrate new AlertMessage component
to live with old one and do not affect each other behaviour.
To achieve this:
- AlertMessage component (new one) has the same set of properties. As result it will allow easily consume old alert messages by new component.
- To preserve old messages without any changes, one additional property is
added to specify which Alert component to show, old or new one.
- Current implementation reuses existing logic where we store state of alert in
local settings instead of redux store.

Release note: None

koorosh committed:
ui: Add localSetting selector and actions for email subscription alert

Create localSetting selector to preserve alert state (open/closed) for
email subscription notification.
Selector is added to the `bannerAlertsSelector` because it is used to
display global alerts.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44856)
<!-- Reviewable:end -->

koorosh committed:
ui: Fix styles for AlertMessage component

And alerts by default is positioned on the top left
and this position is overridden with margin values
to put alert on right side. But after user closes alert,
it's closing with animation and new margins which
make alert jump from right to left side.
To fix this, we need to override margins with !important.

Release note: None

koorosh committed:
ui: Style Close button for Alert panel

Release note: None

dhartunian commented on pkg/ui/src/views/shared/components/alertMessage/alertMessage.tsx:
```diff
@@ -0,0 +1,94 @@
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import React from "react";
```
I'm confused as to which new components go in `src/views/shared/components` and which ones in `/src/components`? I assumed all new work would go in the second directory.

koorosh replied:
@dhartunian , I also thought it has to be in `/src/components`, but this component is implemented in such a way to be compatible (reuse the same interface as another AlertBox component).
This required to use `react-router-dom` lib and reference to `import { AlertInfo, AlertLevel } from "src/redux/alerts"` inside component.
With all of this dependencies it doesn't look like pure component so I decided to put in shared components.

dhartunian commented on pkg/ui/src/views/shared/components/alertMessage/alertMessage.tsx:
```diff
@@ -0,0 +1,94 @@
+};
+
+export class AlertMessage extends React.Component<AlertMessageProps> {
+  static defaultProps: Partial<AlertMessageProps> = {
```
why is this defaultProps necessary?

koorosh replied:
Indeed, not necessary! Has to be cleaned up.

dhartunian commented on pkg/ui/src/redux/alerts.ts:
```diff
@@ -50,6 +51,9 @@ export interface Alert extends AlertInfo {
   dismiss: ThunkAction<Promise<void>, AdminUIState, void>;
+  // Makes alert to be positioned in the top right corner of the screen instead of
+  // stretching to full width.
+  showAsAlert?: boolean;
```
can this be a bit more specific? maybe say `showAsAlertMessage`. This is a bit vague and confusing since the word alert appears everywhere.

dhartunian commented on pkg/ui/src/redux/alerts.spec.ts:
```diff
@@ -350,6 +355,26 @@ describe("alerts", function() {
+        // dismiss alert
+        const alert = emailSubscriptionAlertSelector(state());
+        await alert.dismiss(dispatch, state);
+        const dismissedState = emailSubscriptionAlertLocalSetting.selector(state());
```
Do you think this should also be `openState`? It reads a bit weird to dismiss the alert and then check that `dismissedState` `isFalse`. What do you think?

koorosh replied:
Hah, good point! Thank you!

koorosh committed:
ui: Get rid of useless defaultProps for AlertMessage component

Release note: None

koorosh committed:
ui: Enhance variable namings in tests

Release note: None

dhartunian commented:
bors r+
